packages/eas-cli/src/graphql/mutations/PublishMutation.ts

```diff
@@ -0,0 +1,57 @@
+import publishGraphQL from 'graphql-tag';
+
+import { graphqlClient, withErrorHandlingAsync } from '../client';
+import { PublishUpdateGroupInput, Update } from '../generated';
+
+const PublishMutation = {
+  async getUploadURLsAsync(contentTypes: string[]): Promise<{ specifications: string[] }> {
+    const data = await withErrorHandlingAsync(
+      graphqlClient
+        .mutation<
+          { asset: { getSignedAssetUploadSpecifications: { specifications: string[] } } },
+          { contentTypes: string[] }
+        >(
+          publishGraphQL` 
+        mutation {
+          asset {
+            getSignedAssetUploadSpecifications(assetContentTypes: ${JSON.stringify(contentTypes)}) {
+              specifications
+            }
+          }
+        }`,
+          {
+            contentTypes,
+          }
+        )
+        .toPromise()
+    );
+    return data.asset.getSignedAssetUploadSpecifications;
+  },
+
+  async publishUpdateGroupAsync(
+    publishUpdateGroupInput: PublishUpdateGroupInput
+  ): Promise<Pick<Update, 'updateGroup'>> {
+    const data = await withErrorHandlingAsync(
+      graphqlClient
+        .mutation<
+          { updateRelease: { publishUpdateGroup: Pick<Update, 'updateGroup'>[] } },
+          { publishUpdateGroupInput: PublishUpdateGroupInput }
+        >(
+          publishGraphQL`
+          mutation publish($publishUpdateGroupInput: PublishUpdateGroupInput) {
+            updateRelease {
+              publishUpdateGroup(publishUpdateGroupInput: $publishUpdateGroupInput) {
+                updateGroup
+              }
+            }
+          }
+        `,
+          { publishUpdateGroupInput }
+        )
+        .toPromise()
+    );
+    return data.updateRelease.publishUpdateGroup[0];
```

Comment:
Is there a reason why we only return the first `Update`?

Reply:
We only display the `updateGroup` and `updateMessage` when logging a success error message. These are the same between the updates.

Reply:
If it's not too late to change the GraphQL API, perhaps we could return an object instead of an array? So instead of the current return type `[Update]!`, we would return something like this:
```graphql
type PublishUpdateGroupResult {
  updateGroup: String!
  updates: [Update!]!
}
```
This also makes it possible to add fields to the result later without breaking backwards compatibility (impossible if we return an array).